### ISSUE TITLE
[BUG] accessibility-checker not surfacing network error when parsing config

### ISSUE THREAD

**Issue by jendowns** [IBMa/equal-access#102]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [x] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

> Node version: 11.15.0
> Platform: CircleCI

We use `accessibility-checker` as part of [`@carbon/ibm-security`](https://github.com/carbon-design-system/ibm-security)'s unit test suite. Running those tests in a CI environment (specifically CircleCI) occasionally fails with this console output:

```
undefined:1
undefined
^
 
SyntaxError: Unexpected token u in JSON at position 0
 
    at JSON.parse (<anonymous>)
    at Request._callback (/home/circleci/project/node_modules/accessibility-checker/lib/ACConfigLoader.js:109:34)
    at self.callback (/home/circleci/project/node_modules/accessibility-checker/node_modules/request/request.js:185:22)
    at Request.emit (events.js:193:13)
    at Request.Object.<anonymous>.Request.onRequestError (/home/circleci/project/node_modules/accessibility-checker/node_modules/request/request.js:877:8)
    at ClientRequest.emit (events.js:193:13)
    at TLSSocket.socketErrorListener (_http_client.js:397:9) 
    at TLSSocket.emit (events.js:193:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at processTicksAndRejections (internal/process/task_queues.js:81:17)
```

Digging into that trace, it appears as though  **`ACConfigLoader.js` is [swallowing any server errors it receives](https://github.com/IBMa/equal-access/blob/master/accessibility-checker/src/lib/ACConfigLoader.js#L108) when the operation to fetch `archives.json` fails.**

https://github.com/IBMa/equal-access/blob/495d46caac16cf8e99c4a5854d9ea2e6cdef0ae8/accessibility-checker/src/lib/ACConfigLoader.js#L108-L110

Because nothing is being done with the `error` object in the `request.get()` callback, it is difficult to debug what could be causing our tests to fail.

## Steps to reproduce the issue

I imagine you could mimic our situation by blocking network traffic to the rule server and then trying to run a test suite.

Our repo is [`@carbon/ibm-security`](https://github.com/carbon-design-system/ibm-security) if you need an environment. You could clone our repo and then run `yarn` and then `yarn test` to run our test suite, which includes a matcher with `accessibility-checker`.
